app/queries/lookup_by_base_paths.rb

```diff
@@ -0,0 +1,57 @@
+module Queries
+  module LookupByBasePaths
+    def self.call(base_paths)
+      base_paths_and_content_ids = Edition.with_document
+        .left_outer_joins(:unpublishing)
+        .left_outer_joins(:access_limit)
+        .where(base_path: base_paths)
+        .where("access_limits.edition_id IS NULL")
+        .where("state != 'superseded'")
+        .order(:base_path)
+        .order(:state)
+        .order(user_facing_version: :desc)
+        .pluck(
+          %{
+            DISTINCT ON (editions.base_path, editions.state)
+            editions.base_path,
+            documents.content_id,
+            documents.locale,
+            editions.document_type,
+            CASE editions.state WHEN 'draft' THEN 'draft' ELSE 'live' END,
+            unpublishings.type,
+            unpublishings.alternative_path
+          }
+        )
+
+      lookups = base_paths_and_content_ids.group_by(&:first)
+
+      lookups.each_with_object({}) do |group, result|
```

Comment:
I think so, since you're providing it with a list of things to look up it feels more semantically helpful to return results that correspond to the list you provided